### PR TITLE
pythonPackages.sfepy: 2019.4 -> 2020.4

### PR DESCRIPTION
**Pull request by wd15**
###### Motivation for this change

Update Sfepy to its latest version.

 * Sfepy no longer appears to work with Python 3.7 anymore so turned it off for 3.7.
 * Sfepy has a new dependency on meshio, which I've added. Meshio was added by @Wulfsta in #100135, but that PR appears to be in a zombie state (also #86518 didn't go anywhere). 
 * Updated exdown as meshio requires a later version than the current one on master.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Wulfsta** [comment]
Yeah, sorry about those PRs - as you can tell those packages are extremely low priority to me and I kept getting busy with other things. I’ll close the open one.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 114501` at f7b2b081 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/cf8754f30c6d9fd6bf770888fe137b25">python37Packages.meshio</a></li></li>
    <li><a href="https://gist.github.com/c26874df6946a39921b74b9ba9ea2c94">python38Packages.meshio</a></li></li>
    <li><a href="https://gist.github.com/06f7654e086846338ee705b202411952">python38Packages.sfepy</a></li></li>
    <li><a href="https://gist.github.com/be2cb7943e85826938705896daf0d09d">python39Packages.meshio</a></li></li>
    <li><a href="https://gist.github.com/4620f266d31436f6bf930c07e86eeede">python39Packages.sfepy</a></li></li>
  </ul>
</details>
<details>
  <summary>11 packages built:</summary>
  <ul>
    <li><a href="https://gist.github.com/1677989f7c1161cf99ff4d7ca377e559">python37Packages.dufte</a></li></li>
    <li><a href="https://gist.github.com/5f5ce41ab24e80d7840e42d3fd8446e3">python37Packages.exdown</a></li></li>
    <li><a href="https://gist.github.com/eb6193ab5ac61e32c2069061b75610a4">python37Packages.perfplot</a></li></li>
    <li><a href="https://gist.github.com/1119594e3bf1ab6cdb4fec3302574047">python37Packages.termplotlib</a></li></li>
    <li><a href="https://gist.github.com/982b6dbf56d198f140e046294c1b0dbc">python38Packages.dufte</a></li></li>
    <li><a href="https://gist.github.com/620a30e9e74cc89ba1e5dccce9ce6900">python38Packages.exdown</a></li></li>
    <li><a href="https://gist.github.com/31db82618ef9eb7714c40200e534fa16">python38Packages.perfplot</a></li></li>
    <li><a href="https://gist.github.com/62ece5666603d4bd82c059d29de417c2">python38Packages.termplotlib</a></li></li>
    <li><a href="https://gist.github.com/96e5298881f15827be646959f73ef111">python39Packages.dufte</a></li></li>
    <li><a href="https://gist.github.com/79ed799e413ec01ef7ba8b24c178ff29">python39Packages.exdown</a></li></li>
    <li><a href="https://gist.github.com/e3310d2c256accda08964b61b60b3573">python39Packages.termplotlib</a></li></li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-explicit-check-phase.md">python-explicit-check-phase</a>

Consider using `pytestCheckHook` in `checkInputs` rather than `checkPhase = "pytest";`.

Near pkgs/development/python-modules/meshio/default.nix:32:3:
```
   |
32 |   checkPhase = ''
   |   ^
```
</li>
  </ul>
</details>

**Comment by wd15** [comment]
>       Consider using `pytestCheckHook` in `checkInputs` rather than `checkPhase = "pytest";`.
>       Near pkgs/development/python-modules/meshio/default.nix:32:3:
>       ```
>          |
>       32 |   checkPhase = ''
>          |   ^
>       ```

Fixed.

**Comment by wd15** [comment]
Result of `nixpkgs-review pr 114501` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)                                    │ 
<details>                                                                                                                                │[wd15@loki:~/git/nixpkgs](sfepy-2020.4)$ git diff --staged
  <summary>16 packages built:</summary>                                                                                                  │diff --git a/pkgs/development/python-modules/exdown/default.nix b/pkgs/de
  <ul>                                                                                                                                   │velopment/python-modules/exdown/default.nix
    <li>python37Packages.dufte</li>                                                                                                      │index 17cab3fe55f..bf0a71984c5 100644
    <li>python37Packages.exdown</li>                                                                                                     │--- a/pkgs/development/python-modules/exdown/default.nix
    <li>python37Packages.meshio</li>                                                                                                     │+++ b/pkgs/development/python-modules/exdown/default.nix
    <li>python37Packages.perfplot</li>                                                                                                   │@@ -3,12 +3,12 @@
    <li>python37Packages.termplotlib</li>                                                                                                │ 
    <li>python38Packages.dufte</li>                                                                                                      │ buildPythonPackage rec {
    <li>python38Packages.exdown</li>                                                                                                     │   pname = "exdown";
    <li>python38Packages.meshio</li>                                                                                                     │-  version = "0.7.1";
    <li>python38Packages.perfplot</li>                                                                                                   │+  version = "0.8.5";
    <li>python38Packages.sfepy</li>                                                                                                      │   format = "pyproject";
    <li>python38Packages.termplotlib</li>                                                                                                │ 
    <li>python39Packages.dufte</li>                                                                                                      │   src = fetchPypi {
    <li>python39Packages.exdown</li>                                                                                                     │     inherit pname version;
    <li>python39Packages.meshio</li>                                                                                                     │-    sha256 = "sha256-vnSso3vmPIjX7JX+NwoxguwqwPHocJACeh5H0ClPcUI=";
    <li>python39Packages.sfepy</li>                                                                                                      │+    sha256 = "1ly67whyfn74nr0dncarf3xbd96hacvzgjihx4ibckkc4h9z46bj";
    <li>python39Packages.termplotlib</li>                                                                                                │   };
  </ul>                                                                                                                                  │ 
</details>

File: pkgs/development/python-modules/meshio/default.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  propagatedBuildInputs = [
+    numpy
+    netcdf4
+    h5py
+  ];
+
+  checkInputs = [
+    exdown
+    pytestCheckHook
+  ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/meshio/default.nix:30]:
```suggestion
  ];
  
  pythonImportsCheck = [ "meshio" ];
```

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -7229,7 +7231,7 @@ in {
 
   sexpdata = callPackage ../development/python-modules/sexpdata { };
 
-  sfepy = callPackage ../development/python-modules/sfepy { };
+  sfepy = disabledIf (!pythonAtLeast "3.8") (callPackage ../development/python-modules/sfepy { });
```
Comment (SuperSandro2000) [inline pkgs/top-level/python-packages.nix:7234]:
```suggestion
  sfepy = callPackage ../development/python-modules/sfepy { };
```

File: pkgs/development/python-modules/sfepy/default.nix [hunk 3]
```diff
@@ -9,15 +9,19 @@
...
 , python
 , sympy
+, meshio
+, mpi4py
+, psutil
+, openssh
 }:
 
 buildPythonPackage rec {
   name = "sfepy_${version}";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/sfepy/default.nix:19]:
```suggestion
  pname = "sfepy";
```

File: pkgs/development/python-modules/sfepy/default.nix [hunk 4]
```diff
@@ -9,15 +9,19 @@
...
-  version = "2019.4";
+  version = "2020.4";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/sfepy/default.nix:20]:
```suggestion
  version = "2020.4";
  disabled = pythonOlderThan "3.8";
```

File: pkgs/development/python-modules/sfepy/default.nix [hunk 5]
```diff
@@ -9,15 +9,19 @@
 , cython
 , python
 , sympy
+, meshio
+, mpi4py
+, psutil
+, openssh
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/sfepy/default.nix:15]:
```suggestion
, openssh
, pythonOlderThan
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/top-level/python-packages.nix (hunk 2 above)
Comment (wd15) [reply pkgs/top-level/python-packages.nix]: Done. Makes more sense to keep the logic in `default.nix`.

File: pkgs/development/python-modules/sfepy/default.nix (hunk 3 above)
Comment (wd15) [reply pkgs/development/python-modules/sfepy/default.nix]: Done. No idea why or how that was there.

File: pkgs/development/python-modules/sfepy/default.nix (hunk 4 above)
Comment (wd15) [reply pkgs/development/python-modules/sfepy/default.nix:21]: Done. Using `pythonAtLeast` as I can't see where `pythonOlderThan` is  ever used in Nixpkgs.

File: pkgs/development/python-modules/sfepy/default.nix (hunk 5 above)
Comment (wd15) [reply pkgs/development/python-modules/sfepy/default.nix:15]: Using `pythonAtLeast`.

**Comment by wd15** [comment]
@SuperSandro2000, thanks so much for your work reviewing. Things should be good now.

File: pkgs/development/python-modules/meshio/default.nix (hunk 1 above)
Comment (wd15) [reply pkgs/development/python-modules/meshio/default.nix:30]: Done.

File: pkgs/development/python-modules/sfepy/default.nix (hunk 4 above)
Comment (SuperSandro2000) [reply pkgs/development/python-modules/sfepy/default.nix:21]: Its just pythonOlder. Can't remember the names.

File: pkgs/development/python-modules/sfepy/default.nix [hunk 6]
```diff
@@ -9,15 +9,21 @@
...
+, pythonAtLeast
 }:
 
 buildPythonPackage rec {
-  name = "sfepy_${version}";
-  version = "2019.4";
+  name = "sfepy";
+  version = "2020.4";
+  disabled = !pythonAtLeast "3.8";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/sfepy/default.nix:22]:
```suggestion
  disabled = pythonOlder "3.8";
```

File: pkgs/development/python-modules/sfepy/default.nix [hunk 7]
```diff
@@ -9,15 +9,21 @@
 , cython
 , python
 , sympy
+, meshio
+, mpi4py
+, psutil
+, openssh
+, pythonAtLeast
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/sfepy/default.nix:16]:
```suggestion
, pythonOlder
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Pull request merged**